pkg/resources/validation/cluster/cluster.go

```diff
@@ -93,16 +154,136 @@ func (v *Validator) Admit(request *admission.Request) (*admissionv1.AdmissionRes
 			Allowed: false,
 		}, nil
 	}
+	return psa.AdmissionResponseAllowed(), nil
+}
 
-	return &admissionv1.AdmissionResponse{
-		Allowed: true,
-	}, nil
+// validatePSACT validates the cluster spec when PodSecurityAdmissionConfigurationTemplate is used
+func (v *Validator) validatePSACT(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	oldCluster, newCluster, err := objectsv3.ClusterOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get old and new clusters from request: %w", err)
+	}
+	newTemplateName := newCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	oldTemplateName := oldCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	parsedVersion, err := getClusterVersion(newCluster.Spec.RancherKubernetesEngineConfig.Version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse cluster version: %w", err)
+	}
+	if parsedRangeLessThan125(parsedVersion) && newTemplateName != "" {
+		msg := "PodSecurityAdmissionConfigurationTemplate(PSACT) is only supported in Kubernetes version 1.25 and above"
+		return psa.AdmissionResponseBadRequest(msg), nil
+	}
+	if newTemplateName != "" {
+		response, err := v.checkPSAConfigOnCluster(newCluster)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check the PodSecurity Config in the cluster %s: %w", newCluster.Name, err)
+		}
+		if !response.Allowed {
+			return response, nil
+		}
+	} else {
+		switch request.Operation {
+		case admissionv1.Create:
+			return psa.AdmissionResponseAllowed(), nil
+		case admissionv1.Update:
+			// In the case of unsetting DefaultPodSecurityAdmissionConfigurationTemplateName,
+			// validate that the configuration for PodSecurityAdmission under the kube-api.admission_configuration section
+			// is different between the new and old clusters.
+			// It is possible that user drops DefaultPodSecurityAdmissionConfigurationTemplateName and set the config
+			// under kube-api.admission_configuration at the same time.
+			if oldTemplateName != "" {
+				newConfig, found := psa.GetPluginConfigFromCluster(newCluster)
+				if !found {
+					// not found means the kube-api.admission_configuration section is also removed, which is good
+					return psa.AdmissionResponseAllowed(), nil
+				}
+				oldConfig, _ := psa.GetPluginConfigFromCluster(oldCluster)
+				if reflect.DeepEqual(newConfig, oldConfig) {
+					msg := "The Plugin Config for PodSecurity under kube-api.admission_configuration is the same as the previously-set PodSecurityAdmissionConfigurationTemplate." +
```

Comment:
I'm not sure why we should not allow setting `kube-api.admission_configuration` explicitly to the same value as the configuration per previously set PSACT. Maybe users just want to "untie" from the configuration template.
Maybe I'm missing a requirement here.

Reply:
please see my reply [here](https://github.com/rancher/webhook/pull/165#discussion_r1061045288)